File: backend/app/models/resume.py
"""
文件名: app/models/resume.py
创建时间: 2026-06-26
作者: TalentSense Team
功能描述: 简历模型，对应 API-Design.md 10.1/10.2
"""
from pydantic import BaseModel, Field, field_validator


class Salary(BaseModel):
    min: int
    max: int


class BasicInfo(BaseModel):
    name: str
    phone_masked: str
    email_masked: str
    gender: str | None = None
    age: int | None = None
    location: str | None = None


class WorkExperience(BaseModel):
    company: str
    position: str
    start_date: str
    end_date: str
    description: str


class EducationDetail(BaseModel):
    school: str
    major: str
    degree: str
    start_date: str
    end_date: str


class FileInfo(BaseModel):
    file_name: str
    file_type: str
    file_size: int | None = None


class ParseInfo(BaseModel):
    parse_status: str
    parse_time: str | None = None


class ResumeListItem(BaseModel):
    resume_id: str
    candidate_id: str
    name: str
    gender: str | None = None
    age: int | None = None
    education: str
    education_level: int = Field(ge=0, le=3)
    work_years: int
    skills: list[str] = []
    expected_salary: Salary
    tags: list[str] = []
    is_favorite: bool = False
    parse_status: str
    location: str | None = None
    created_at: str

    @field_validator("education_level")
    @classmethod
    def check_level(cls, v):
        if v not in (0, 1, 2, 3):
            raise ValueError("education_level must be 0-3")
        return v


class ResumeDetail(ResumeListItem):
    basic_info: BasicInfo
    work_experience: list[WorkExperience] = []
    education_detail: list[EducationDetail] = []
    summary: str = ""
    file_info: FileInfo | None = None
    parse_info: ParseInfo | None = None
    notes: str = ""
    interview_notes: list = []
    updated_at: str | None = None


class UploadResponse(BaseModel):
    resume_id: str
    candidate_id: str
    file_name: str
    parse_status: str = "parsing"
    is_duplicate: bool = False
    duplicate_with: str | None = None


class PreviewResponse(BaseModel):
    preview_url: str
    file_type: str
    expires_in: int

File: frontend/src/types/resume.ts
/**
 * 文件名: types/resume.ts
 * 创建时间: 2026-06-26
 * 作者: TalentSense Team
 * 功能描述: 简历类型，对应 API-Design.md 2.0 / 第十章
 */

export interface ResumeListItem {
  resume_id: string
  candidate_id: string
  name: string
  gender?: string
  age?: number
  education: string
  education_level: number
  work_years: number
  skills: string[]
  expected_salary: { min: number; max: number }
  tags: string[]
  is_favorite: boolean
  parse_status: 'pending' | 'parsing' | 'completed' | 'failed'
  location?: string
  created_at: string
  // RBAC 脱敏字段：admin 可见原始值，普通用户可见 masked
  phone?: string
  email?: string
  phone_masked?: string
  email_masked?: string
  summary?: string
}

export interface WorkExperience {
  company: string
  position: string
  start_date: string
  end_date: string
  description: string
}

export interface EducationDetail {
  school: string
  major: string
  degree: string
  start_date: string
  end_date: string
}

export interface ProjectItem {
  name: string
  role: string
  description: string
}

export interface InterviewNoteItem {
  note_id: string
  interviewer: string
  rating: number
  result: string
  content: string
  created_at: string
}

export interface ResumeDetail extends ResumeListItem {
  basic_info: {
    name: string
    phone_masked: string
    email_masked?: string
    gender?: string
    age?: number
    location?: string
  }
  work_experience: WorkExperience[]
  education_detail: EducationDetail[]
  projects?: ProjectItem[]
  summary: string
  file_info: { file_name: string; file_type: string; file_size?: number }
  parse_info: { parse_status: string; parse_time?: string }
  notes: string
  interview_notes?: InterviewNoteItem[]
  updated_at?: string
}

export interface UploadResponse {
  resume_id: string
  candidate_id: string
  file_name: string
  parse_status: string
  is_duplicate: boolean
  duplicate_with: string | null
}

export interface ResumeListQuery {
  page?: number
  page_size?: number
  keyword?: string
  tag?: string
  is_favorite?: boolean
  education_min?: number
  work_years_min?: number
  salary_min?: number
  salary_max?: number
  status?: string
  // 筛选和排序（新增）
  date_from?: string
  date_to?: string
  sort_by?: string
  sort_order?: 'asc' | 'desc'
}
